# HG changeset patch
# User Michael Godfrey <michaeldgodfrey@gmail.com>
# Date 1360436193 18000
#      Sat Feb 09 13:56:33 2013 -0500
# Node ID 7398b2dd08cb84b08b50738ef498d48e9fec12be
# Parent  7285f607b3835c485c061fee425520a1b4970ddf
stemleaf.m: Correct signbit test to use != 0 (bug #38291).

* scripts/plot/stemleaf.m: Correct signbit test to use != 0.

diff --git a/scripts/plot/stemleaf.m b/scripts/plot/stemleaf.m
--- a/scripts/plot/stemleaf.m
+++ b/scripts/plot/stemleaf.m
@@ -198,17 +198,17 @@ function varargout = stemleaf (x, stem_u
   for kx = 2:nstems
     line_out = "";
     steml    = "";
     ## Build a string of leaf digits for stem(kx) if stem(kx) <= 0, or
     ## stem(kx-1) if stem(kx) > 0
 
     ## stems -+ 0 have to be handled as special cases.
     for xi = 1:nx
-      if(signbit(stems(kx)) == 1)
+      if(signbit(stems(kx)) != 0)
         t1 = ((x(xi) <= stems(kx)*10) && (x(xi) > (stems(kx-1)*10)));
       else
         t1 = ((x(xi) < stems(kx)*10) && (x(xi) >= (stems(kx-1)*10)));
       endif
       ## Special tests for stem -+ 0
       if ((stems(kx) == 0) && signbit(stems(kx)) && (x(xi) == 0)) && !signbit(x(xi))
         t1 = 0;
       endif
